### ISSUE TITLE
java.lang.reflect.InvocationTargetException 

### ISSUE THREAD

**Issue by kingerking** [TomasMikula/RichTextFX#432]
I keep getting a java.lang.reflect.InvocationTargetException when I just try to launch an application using any of the Classes from RitchTextFX. 
Here is my stack if anyone wants it:

```
Executing C:\Users\Kyles PC\Documents\NetBeansProjects\JavaFXApplication1\dist\run1557054747\JavaFXApplication1.jar using platform C:\Program Files\Java\jdk1.8.0_101\jre/bin/java
Exception in Application start method
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(LauncherImpl.java:389)
	at com.sun.javafx.application.LauncherImpl.launchApplication(LauncherImpl.java:328)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:767)
Caused by: java.lang.RuntimeException: Exception in Application start method
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:917)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$155(LauncherImpl.java:182)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: org/reactfx/value/SuspendableVal
	at javafxapplication1.JavaFXApplication1.start(JavaFXApplication1.java:44)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$162(LauncherImpl.java:863)
	at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$175(PlatformImpl.java:326)
	at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
	... 1 more
Caused by: java.lang.ClassNotFoundException: org.reactfx.value.SuspendableVal
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 10 more
Exception running application javafxapplication1.JavaFXApplication1
Java Result: 1
```

**Comment by afester** [comment]
This is caused by 

`java.lang.ClassNotFoundException: org.reactfx.value.SuspendableVal`

Do you have the ReactFX jar files included in your project?

**Comment by kingerking** [comment]
Yup. Tried it with eclipse and nebeans Incase it was a bug

On Jan 19, 2017 11:21 PM, "Andreas" <notifications@github.com> wrote:

This is caused by

java.lang.ClassNotFoundException: org.reactfx.value.SuspendableVal

Do you have the ReactFX jar files included in your project?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/TomasMikula/RichTextFX/issues/432#issuecomment-273987142>,
or mute the thread

**Comment by afester** [comment]
Are you launching `JavaFXApplication1.jar` as a standalone, self-contained, runnable jar file? If yes, have you verified that the jar file contains all classes from the dependent jars? Otherwise, how are you launching the application? And how are you creating `JavaFXApplication1.jar`?

**Comment by kingerking** [comment]
I'm running it within the Netbeans IDE. how can I check if the library is being included? I have used NetBeans with 3rd party libraries before and have had no issues with this sort of thing so I don't think I'm including it wrong.

**Comment by kingerking** [comment]
I have concluded that I am doing nothing wrong regarding my installation method, I have installed a different API (org.json) and it work's perfectly before and after export. also all the classes are in the exported jar.

**Comment by JordanMartinez** [comment]
@kingerking Just to clarify, how are you getting the RichTextFX Jar? It seems that you are not using Gradle. Did you manually download it? If so, did you download the fat jar or the "skinny" one?

**Comment by kingerking** [comment]
I'm getting it from the link in the readme on the front page of the repository. And whether I use fat or "skinny" it doesn't matter since I've tested with both of them. they both produce identical error's

**Comment by JordanMartinez** [comment]
Closed by #436. (I linked to the wrong issue when I fixed it)
